ql/src/java/org/apache/hadoop/hive/ql/exec/repl/incremental/IncrementalLoadTasksBuilder.java

```diff
@@ -163,6 +163,17 @@ public IncrementalLoadTasksBuilder(String dbName, String loadPath,
       taskChainTail.addDependentTask(updateIncPendTask);
       taskChainTail = updateIncPendTask;
 
+      Database dbToLoadIn = hive.getDatabase(dbName);
+      if (!ReplUtils.isTargetOfReplication(dbToLoadIn)) {
+        Map<String, String> props = new HashMap<>();
+        props.put(ReplUtils.TARGET_OF_REPLICATION, "true");
+        AlterDatabaseSetPropertiesDesc setTargetDesc = new AlterDatabaseSetPropertiesDesc(dbName, props, null);
+        Task<?> addReplTargetPropTask = TaskFactory.get(new DDLWork(inputs, outputs, setTargetDesc, true,
+                dumpDirectory, metricCollector), conf);
+        taskChainTail.addDependentTask(addReplTargetPropTask);
+        taskChainTail = addReplTargetPropTask;
```

Comment:
Shouldn't this be done at the beginning of incremental load and not at the end. Else, during controlled failover, the events will be loaded first and then only this property would be set. If there is bootstrap dump attempted on src, that would pass.

Reply:
Done